blockchain/proxy.go

```diff
@@ -820,23 +826,21 @@ func resolveBlockRange(filter *sdkCore.FilterQuery, latestBlock uint64) (uint64,
 }
 
 // getBlockNumberByHash retrieves the block number associated with the given block hash
-func (ep *proxy) getBlockNumberByHash(ctx context.Context, shardID uint32, blockHash *[32]byte) (uint64, error) {
-	blockHashStr := hex.EncodeToString(blockHash[:])
-
-	buff, code, err := ep.GetHTTP(ctx, ep.endpointProvider.GetBlockByHash(shardID, blockHashStr))
-	if err != nil || code != http.StatusOK {
-		return 0, createHTTPStatusError(code, err)
+func (ep *proxy) getBlockNumberByHash(ctx context.Context, shardID uint32, blockHash []byte) (uint64, error) {
+	blockHashStr := string(blockHash)
```

Comment:
`hex.EncodeToString` should remain, otherwise the converted []byte to string directly might contain unreadable characters and the next API call will most certainly fail.